### ISSUE TITLE
MySQL Database error upon generating one-time code

### ISSUE THREAD

**Issue by HigH-HawK** [ChristophWurst/twofactor_admin#44]
Hi,

first of all this app is awesome as it helps a lot. We all know that the "average user" does not want to get involved in technical stuff. All they want is, that things are working.

I installed the app today on our company DEV NC system and did some tests. I immediately noticed that there seems to be a problem with the newly created table in the database.

When issuing the command `sudo -u apache php occ twofactorauth:admin:generate-code username` I get an SQL error message as follows:

```
In AbstractMySQLDriver.php line 122:

  An exception occurred while executing 'INSERT INTO `oc_twofactor_admin_codes` (`user_id`, `code`, `expires`) VALUES(?, ?, ?)' with params ["username", "205369", 1559907659]:

  SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value


In PDOStatement.php line 143:

  SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value


In PDOStatement.php line 141:

  SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value


twofactorauth:admin:generate-code <uid>
```

I went to check in the database and noticed that the error message is right, the field `id` does not have a default value nor an auto_increment (which wouldn't make any sense anyway)

What I did to fix this error message, I just added a default value of 0 to the field `id` in the database and could execute the above command without any further issues. Maybe this could be fixed in the next version.

Versions
Nextcloud: 16.0.1
2FA Admin: 0.3.0

**Comment by kierankyllo** [comment]
Great insight @HigH-HawK 

Thank You

In case its helpful for anyone, the following commands worked for me to correct the issue.

sudo -s
mysql -u root -p
USE nextcloud;
DESCRIBE oc_twofactor_admin_codes;
#confirm default is NULL
ALTER TABLE oc_twofactor_admin_codes
ALTER id SET DEFAULT '0';
#confirm new default value
DESCRIBE oc_twofactor_admin_codes;
quit
service mysql reload
systemctl restart mariadb

**Comment by ChristophWurst** [comment]
What version of mysql do you use?

**Comment by HigH-HawK** [comment]
In my case, this is the version of MySQL / MariaDB I am using:

Output from `mysql -V`

```
mysql  Ver 15.1 Distrib 10.3.15-MariaDB, for Linux (x86_64) using readline 5.1
```

**Comment by kierankyllo** [comment]
Hi Chris,

mysql  Ver 15.1 Distrib 10.3.15-MariaDB, for debian-linux-gnu (x86_64)
using readline 5.2

Kyllo

**Comment by HigH-HawK** [comment]
I updated to version 0.4.0 today and tried generating a one-time code via the Nextcloud web interface. 

I am getting the error `Could not generate a code: unknown error`

I looked into the Nextcloud logs and found the following detailed error

```
Doctrine\DBAL\Exception\NotNullConstraintViolationException: An exception occurred while executing 'INSERT INTO `oc_twofactor_admin_codes` (`user_id`, `code`, `expires`) VALUES(?, ?, ?)' with params ["username", "968421", 1561733614]: SQLSTATE[HY000]: General error: 1364 Field 'id' doesn't have a default value
```

The field `id` is again missing the default value. After setting a default value (0 in my case), I can generate a code without any issues.

Output from `mysql -V`

```
mysql  Ver 15.1 Distrib 10.3.15-MariaDB, for Linux (x86_64) using readline 5.1
```

**Comment by szaimen** [comment]
Any update on this?
I've got the same error.

**Comment by ChristophWurst** [comment]
Anybody using v0.4.1 of this app yet?

**Comment by HigH-HawK** [comment]
Hey @ChristophWurst 

I did update mine to v0.4.1 last week but totally forgot to report, sorry about that. After the update I generated a one time code to test the update and it worked fine without any errors.

I also used a fresh NC 16.0.4 installation which never had this app installed and installed the new v0.4.1 - I could generate a one time without any problems as well, so for me this issue can be closed as fixed in v0.4.1 :+1:
